### PR TITLE
Change Vertices.indices to use a Uint16 list to more accurately reflect Skia's API

### PR DESCRIPTION
**Pull request by jason-simmons**
Also throw if SkVertices::Builder reports an invalid configuration.

Fixes https://github.com/flutter/flutter/issues/31270

### CODE REVIEW

File: lib/ui/painting/vertices.cc [hunk 1]
```diff
@@ -77,10 +82,14 @@ void Vertices::init(SkVertices::VertexMode vertex_mode,
     DecodeInts<SkColor>(colors, builder.colors());
   }
 
-  if (indices.data())
-    DecodeInts<uint16_t>(indices, builder.indices());
+  if (indices.data()) {
+    std::copy(indices.data(), indices.data() + indices.num_elements(),
+              builder.indices());
+  }
```
Comment (chinmaygarde) [inline lib/ui/painting/vertices.cc]: Return `false` if the data is not present?

Comment (jason-simmons) [reply lib/ui/painting/vertices.cc]: `indices` is an optional argument

**Comment by chinmaygarde** [comment]
The licenses need patching.

**Comment by chinmaygarde** [comment]
Is this good to go?

**Pull request merged**